slacker/slack_api.py

```diff
@@ -148,3 +161,36 @@ def __validate_response(self, response):
         error = data["error"]
       raise SlackAPIException("Unsuccessful API request: {}\nError: {}"
                               .format(response.url, error), error)
+
+  def __is_cached(self, key):
+    """Look up request in cache given hash key"""
+    if self.__cache is None:
+      return False
+
+    if self.__cache.get(key) is not None:
+      self.__logger.debug("Cache hit: {}".format(key))
+      return True
+    self.__logger.debug("Cache miss: {}".format(key))
+    return False
+
+  def __get_cached_value(self, key):
+    """Get value from cache given the hash key"""
+    if self.__cache is None:
+      return None
+    return self.__cache.get(key)
+
+  def __update_cache(self, key, resp):
+    """Update cache value given a key"""
+    self.__logger.debug("Updating cache: {}".format(key))
+    self.__cache[key] = resp
+    return self.__cache[key]
+
+  def __generate_cache_key(self, url, params):
+    """Returns hash of the url and params"""
+    if self.__cache is None:
+      return None
+
+    m = hashlib.md5()
```

Comment:
I'd prefer to use a SHA family hash instead, like SHA-256 - it doesn't have to be overkill.